src/FGame.cpp

```diff
@@ -108,15 +105,20 @@ void FGame::render()
 		}
 	}
 			
-	// Render characters
+	// Render characters and their weapons
 	for (auto &character : m_engine->getCharacters())
 	{
-		auto sprite = loadSprite(m_characterTypeMap[character.getType()], character.getSize().x, character.getSize().y);
-		sprite->setPosition(sf::Vector2f(character.getPosition().x, character.getPosition().y));
-		m_window->draw(*sprite);
-		
-		// Render current weapon
-		//TODO
+		if (character.getHealth() > 0)
+		{
+			auto sprite = loadSprite(m_characterTypeMap[character.getType()], character.getSize().x, character.getSize().y);
+			sprite->setPosition(sf::Vector2f(character.getPosition().x, character.getPosition().y));
+			m_window->draw(*sprite);
+			
+			// auto weapon = character.getWeapon();
+			// auto weaponSprite = loadSprite(m_weaponTypeMap[weapon.getType()], weapon.getSize().x, weapon.getSize().y);
+			// weaponSprite->setPosition(sf::Vector2f(character.getPosition().x + character.getSize().x / 2.f, character.getPosition().y + character.getSize().y / 2.f));
+			// m_window->draw(*weaponSprite);
```

Comment:
Oh nice. If you want, PR away. :)


Med vänliga hälsningar

Marcus Näslund
naslundx@gmail.com
www.kvadratrot.se

On Tue, Apr 26, 2016 at 2:08 PM, Deniz Kennedy <notifications@github.com>
wrote:

> In src/FGame.cpp
> <https://github.com/naslundx/FightR2/pull/15#discussion_r61073632>:
>
> > -		auto sprite = loadSprite(m_characterTypeMap[character.getType()], character.getSize().x, character.getSize().y);
> > -		sprite->setPosition(sf::Vector2f(character.getPosition().x, character.getPosition().y));
> > -		m_window->draw(*sprite);
> > -		
> > -		// Render current weapon
> > -		//TODO
> > +		if (character.getHealth() > 0)
> > +		{
> > +			auto sprite = loadSprite(m_characterTypeMap[character.getType()], character.getSize().x, character.getSize().y);
> > +			sprite->setPosition(sf::Vector2f(character.getPosition().x, character.getPosition().y));
> > +			m_window->draw(*sprite);
> > +			
> > +			// auto weapon = character.getWeapon();
> > +			// auto weaponSprite = loadSprite(m_weaponTypeMap[weapon.getType()], weapon.getSize().x, weapon.getSize().y);
> > +			// weaponSprite->setPosition(sf::Vector2f(character.getPosition().x + character.getSize().x / 2.f, character.getPosition().y + character.getSize().y / 2.f));
> > +			// m_window->draw(*weaponSprite);
>
> Checked and got the bug. In the file FCharacter.cpp, the variable
> m_currentWeapon does not get any value (hence some garbage value, but the
> variable is already defined in the header file :) ) and it is used as an
> index to the vector of weapons. When I put the line "m_currentWeapon = 0"
> into the constructors in FCharacter.cpp, it works like charm ;)
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/naslundx/FightR2/pull/15/files/a908ed267e9404f1ca334453535bfe8ae5845581#r61073632>
>